content/articles/data-visualization-in-react-using-apexcharts/index.md

```diff
@@ -0,0 +1,260 @@
+  const [averageTemp, setAverageTemp] = useState([]);
+  const [date, setDate] = useState([]);
+
+  useEffect(() => {
+    const getData = async () => {
+    const url = 'http://localhost:8000/temperature';
+    try {
+      const response = await fetch(url);
+      const data = await response.json();
+      console.log(data);
+      setAverageTemp(data?.map((item) => item.average_temp));
+      setDate(data?.map((item) => item.date));
+    } catch (error) {
+        console.log(error);
+    }
+  };
+    getData();
+  }, []);
+
+ const series = [ //Data on the y-axis
+    {
+      name: "Temperature in Celsius",
+      data: averageTemp
+    }
+  ];
+  const options = { //data on the x-axis
+  chart: { id: 'bar-chart'},
+  xaxis: {
+    categories: date
+  }
+};
+
+```
+In the code snippet above, useEffect fetches the data from the API endpoint when the component renders the first time. We then assign an API endpoint to the `url` variable. In our example, the endpoint is `http://localhost:8000/temperature`. This is because, we are using a fake REST API from the JSON server, but it can be replaced with any other real API endpoint.
+
+Once the API is called, the response is converted into JSON and stored in a variable `data`. To get the values for our x-axis and y-axis, we map through `data` and return an array of `average_temp`  values. We then store this array in the `averageTemp` state using `setAverageTemp` function.
+We also map through `data` and return the `date` values. This is then stored in state, `date` using the `setDate` function.
+After fetching data using useEffect, `series[0].data` is assigned the `averageTemp` array and `options.xaxis.categories` is assigned `date` array.
+
+ApexCharts.js has various charts as mentioned above. Therefore, if you would like to change the type of chart, you can pass a different chart type to the type props. In the example below, the second chart component takes in a type of `line`.
+
+```javascript
+return (
+  <div>
+    <Chart
+      options={options}
+      series={series}
+      type="bar"
+      width="450"
+    />
+    <Chart
+      options={options}
+      series={series}
+      type="line"
+      width="450"
+    />
+  </div>
+)
+```
+Finally, import the `MyCharts` component into App.js. If you run the project on `localhost:3000`, below is an image of what will be on the browser.
+```javascript
+import MyCharts from './components/charts';
+
+function App() {
+  return (
+    <div className="App">
+      <MyCharts />
+    </div>
+  );
+};
+
+export default App;
+```
+
+![charts](/engineering-education/data-visualization-in-react-using-apexcharts/charts.png)
+
+The first chart is a bar chart, the second chart is a line graph with data from our API endpoint.
+
+### Conclusion
+In this article, we have installed the ApexCharts.js library in our React application, created a fake REST API using a JSON server, fetched data from the API endpoint, and displayed it in a bar chart and line graph.
```

Comment:
A good conclusion statement should always:
- Leave your readers with something to think about.
- Summarize the key supporting ideas you discussed throughout the article. 
- Leave the reader with a solution, an insight, questions for further study or a call to action.